dbms/src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -253,6 +258,47 @@ class AggregateFunctionSumMap final : public IAggregateFunctionDataHelper<
     }
 
     const char * getHeaderFilePath() const override { return __FILE__; }
+
+    virtual bool keepKey(const T & key) const = 0;
+};
+
+template <typename T>
+class AggregateFunctionSumMap final : public AggregateFunctionSumMapBase<T, AggregateFunctionSumMap<T>>
+{
+public:
+    AggregateFunctionSumMap(const DataTypePtr & keys_type, DataTypes & values_types)
+        : AggregateFunctionSumMapBase<T, AggregateFunctionSumMap<T>>{keys_type, values_types}
+    {}
+
+    String getName() const override { return "sumMap"; }
+
+    bool keepKey(const T &) const override { return true; }
+};
+
+template <typename T>
+class AggregateFunctionSumMapFiltered final : public AggregateFunctionSumMapBase<T, AggregateFunctionSumMapFiltered<T>>
+{
+private:
+    std::vector<T> keys_to_keep;
+
+public:
+    AggregateFunctionSumMapFiltered(const DataTypePtr & keys_type, const DataTypes & values_types, const Array & keys_to_keep)
+        : AggregateFunctionSumMapBase<T, AggregateFunctionSumMapFiltered<T>>{keys_type, values_types}
+    {
+        this->keys_to_keep.reserve(keys_to_keep.size());
```

Comment:
`this->` qualification looks unneeded.

Reply:
It is not, because `keys_to_keep` is both a member and an argument to the constructor. It would be fine if this initialization was happening in the initializer list. Should I rename the argument `keys_to_keep_` to avoid the need for `this->`

Reply:
I went and change the argument name for `keys_to_keep_` in order to avoid the explicit `this->`.